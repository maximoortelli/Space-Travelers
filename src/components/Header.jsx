import React from 'react';
import { NavLink } from 'react-router-dom';
import Logo from '../assets/planet.png';
import '../styles/header.css';

const Header = () => {
  const links = [
    { path: '/', text: 'Rockets' },
    { path: 'missions', text: 'Missions' },
    { path: 'profile', text: 'My Profile' },
  ];

  return (
    <div>
      <header className="header">
        <div>
          <img src={Logo} alt="logo" />
          <h1>Space Traveler&apos;s Hub</h1>
        </div>
        <nav>
          <ul>
            {links.map((link) => (
              <li key={link.text.toLowerCase()}>
                <NavLink to={link.path}>{link.text}</NavLink>
              </li>
            ))}
          </ul>
        </nav>
      </header>
    </div>
  );
};

export default Header;
