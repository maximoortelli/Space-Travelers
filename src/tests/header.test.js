import renderer from 'react-test-renderer';
import React from 'react';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import Header from '../components/Header';

test('Confirm that the header is displayed correctly', () => {
  const component = renderer.create(
    <MemoryRouter>
      <Header />
    </MemoryRouter>,
  );

  const headerTree = component.toJSON();
  expect(headerTree).toMatchSnapshot();
});
