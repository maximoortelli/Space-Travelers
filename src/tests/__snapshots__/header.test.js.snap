// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm that the header is displayed correctly 1`] = `
<div>
  <header
    className="header"
  >
    <div>
      <img
        alt="logo"
        src="planet.png"
      />
      <h1>
        Space Traveler's Hub
      </h1>
    </div>
    <nav>
      <ul>
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            className=""
            href="/missions"
            onClick={[Function]}
          >
            Missions
          </a>
        </li>
        <li>
          <a
            className=""
            href="/profile"
            onClick={[Function]}
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
